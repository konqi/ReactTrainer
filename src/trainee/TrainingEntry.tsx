import {
  createStyles,
  Grid,
  makeStyles,
  Paper,
  Theme,
  Typography,
} from '@material-ui/core'
import {format} from 'date-fns'
import de from 'date-fns/locale/de'
import React from 'react'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      border: '2px solid black',
      borderTop: '12px solid black',
      borderRadius: 5,
      boxShadow: theme.shadows[4],
    },
    calendar: {
      alignContent: 'center',
    },
    center: {
      alignSelf: 'center',
    },
  })
)

export interface ExternalProps {
  date: Date
}

export const TrainingEntry: React.FC<ExternalProps> = ({date}) => {
  const classes = useStyles()
  const day = format(date, 'D', {locale: de})
  const month = format(date, 'MMMM', {locale: de})

  return (
    <Paper className={classes.root}>
      <Grid container direction="column" className={classes.calendar}>
        <Grid item className={classes.center}>
          <Typography variant="h4" component="span">
            {day}
          </Typography>
        </Grid>
        <Grid item>{month}</Grid>
      </Grid>
    </Paper>
  )
}
