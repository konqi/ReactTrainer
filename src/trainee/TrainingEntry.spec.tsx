import React from 'react'
import {render} from '@testing-library/react'
import {TrainingEntry} from './TrainingEntry'

const OriginalDate = Date
describe('snapshot tests', () => {
  beforeAll(() => {
    jest.mock('MockDate')
    const {MockDate} = require('MockDate')
    global.Date = MockDate
  })

  afterAll(() => {
    global.Date = OriginalDate
  })

  it('should show a valid entry', () => {
    const {baseElement, unmount} = render(<TrainingEntry date={new Date()} />)
    expect(baseElement).toMatchSnapshot()
    unmount()
  })
})
