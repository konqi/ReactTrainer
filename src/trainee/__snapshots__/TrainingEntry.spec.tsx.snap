// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests should show a valid entry 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root makeStyles-calendar-2 MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root makeStyles-center-3 MuiGrid-item"
        >
          <span
            class="MuiTypography-root MuiTypography-h4"
          >
            21
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          Juli
        </div>
      </div>
    </div>
  </div>
</body>
`;
